### ISSUE TITLE
Insert link warning (button block) and usability - consistency

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1670]
Not sure what are the plans for the insert link "inline toolbar" or if this is already on the radar, so please close this issue if it's pointless.

Noticed two issues:

**Button block: inserting the URL triggers a React warning:**
As soon as I type the first character in the URL field:

![screen shot 2017-07-03 at 16 48 44](https://user-images.githubusercontent.com/1682452/27797890-94acd2d4-600f-11e7-8061-4ea818c99e8d.png)

```
Warning: A component is changing an uncontrolled input of type url to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component.
More info: https://fb.me/react-controlled-components
in input (created by edit)
```

Looks similar to #680

**Usability/consistency:**
When inserting a link on some selected text, the inline toolbar "Apply" button changes to an "Edit" button. The URL displayed in the toolbar is clickable and points to the linked resource, and clicking on "Edit" makes the URL editable again:

<img width="793" alt="screen shot 2017-07-03 at 16 18 37" src="https://user-images.githubusercontent.com/1682452/27798057-297517c8-6010-11e7-8def-06a07e0ffc9c.png">

Instead, on the button block the behavior is different: after a link is inserted, the "Apply" button doesn't change. There's no "Edit" button and the URL displayed in the toolbar is always editable:

<img width="448" alt="screen shot 2017-07-03 at 16 43 29" src="https://user-images.githubusercontent.com/1682452/27798148-7fbd0c80-6010-11e7-828e-6f9842ebe31f.png">

Not sure if there are technical reasons for this. ideally, the behavior should be similar. I can understand the button block doesn't necessarily need a "Remove link" button but I'd say the insert/edit user experience should be as similar as possible with the one of standard links.

**Comment by iseulde** [comment]
+1 You should also be able to navigate to the page. :)

**Comment by afercia** [comment]
Yep! Also, editing an already inserted link with keyboard only is a bit hard right now. I guess that will improve with the introduction of a Cmd/Ctrl+K shortcut together with the other planned shortcuts, see #71

**Comment by mtias** [comment]
We have changed how the button urlInput looks, but it should also be more consistent functionality wise. Feel free to reopen if there are still concerns.

**Comment by afercia** [comment]
@mtias sorry maybe I'm missing something but I don't see relevant changes here. The button link inline toolbar and the normal link inline toolbar still work in a different way and would benefit from some better consistency.

Also, the JS warning is still there:
`A component is changing an uncontrolled input of type text to be controlled.`

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1520531643000136)._

**Comment by afercia** [comment]
@karmatosed I see there are still at least 3 different types of UI to insert a link, see the ones on the blocks. I don't see a reason why they should be different, unless I'm missing something.

cover image:

<img width="575" alt="cover block insert link UI" src="https://user-images.githubusercontent.com/1682452/38744245-8f214d06-3f41-11e8-95b6-9ea2d9866d2b.png">

image:

<img width="363" alt="image block insert link UI" src="https://user-images.githubusercontent.com/1682452/38744246-8f4d5a5e-3f41-11e8-813d-17f7bc061091.png">

paragraph (partially hidden by the following image block):

<img width="435" alt="paragraph block insert link UI" src="https://user-images.githubusercontent.com/1682452/38744247-8f703222-3f41-11e8-83f7-2054735fb5c5.png">

button

<img width="374" alt="button block insert link UI" src="https://user-images.githubusercontent.com/1682452/38744248-8f9759ec-3f41-11e8-80aa-1de03b041b03.png">

**Comment by karmatosed** [comment]
@afercia I totally agree in unifying them to the new UI and we can using this as a starting point: https://github.com/WordPress/gutenberg/issues/6107

**Comment by karmatosed** [comment]
Unless I am wrong we should close this in light of the issue above. We can always reopen but lets make sure we get all the link interfaces unified there.

**Comment by afercia** [comment]
Sure. For reference: https://github.com/WordPress/gutenberg/issues/6392
